### PR TITLE
Add onboarding theme rest api endpoint stub

### LINKED ISSUE #40482: Recommended Themes REST API Endpoint Stub
## User Story

> As a new merchant, I want professionally designed themes recommended to me, so I have an alternative to using the pattern assembler.

## Acceptance Criteria

```
Scenario: Endpoint request paramaters
    Given there is a REST API endpoint
    And the available parameters are
      industry (string): limits the results to themes relevant for this industry (optional)
      currency (enum: USD, AUD, CAD, EUR, GBP): returns pricing in this currency (optional, default: USD)

Scenario: Endpoint response format
    Given there is a REST API endpoint
    And the response should return static content for the following 4 themes: Tsubaki, Tazza, Amulet, Zaino (note: these are Dotcom themes)
    And each theme should include fields for
       Theme Name (string): The name of the theme.
       Price: (string): Human readable price. Examples: Free, $49 / per year, €49 / per year.
       Color Palettes (array): Array of color palette objects, each containing primary, secondary, and background colors.
       Thumbnail URL (string): URL of the thumbnail image.
       Link URL: URL for the theme.

Scenario: Endpoint fallback content
    Given there is a REST API endpoint
    And the user has marketplace suggestions disabled
    Then return 0 themes

Scenario: 3PD Extensibility
    Given there is a REST API endpoint
    And the response should be filterable to allow the theme collection to be customized by third parties.
    And the response should be filterable to allow the "Browse All" URL to be customized by third parties.
    
Scenario: Browse All URL
    Given there is a REST API endpoint
    And the "Browse All" URL should be provided in the _links response

```

## Additional Info

* This is just a "stub" containing static content. A follow-up issue will be created to hook it up with live data from [WooCommerce.com Theme Store](https://woocommerce.com/product-category/themes/?categoryIds=1033&collections=theme&page=1) (marketplace)
* Woo Express will utilize the 3PD filter to provide themes relevant for that environment.

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40482.

Implement `/wc-admin/onboarding/themes/recommended` endpoint stub.

```
Scenario: Endpoint request paramaters
    Given there is a REST API endpoint
    And the available parameters are
      industry (string): limits the results to themes relevant for this industry (optional)
      currency (enum: USD, AUD, CAD, EUR, GBP): returns pricing in this currency (optional, default: USD)

Scenario: Endpoint response format
    Given there is a REST API endpoint
    And the response should return static content for the following 4 themes: Tsubaki, Tazza, Amulet, Zaino (note: these are Dotcom themes)
    And each theme should include fields for
       Theme Name (string): The name of the theme.
       Price: (string): Human readable price. Examples: Free, $49 / per year, €49 / per year.
       Color Palettes (array): Array of color palette objects, each containing primary, secondary, and background colors.
       Thumbnail URL (string): URL of the thumbnail image.
       Link URL: URL for the theme.

Scenario: Endpoint fallback content
    Given there is a REST API endpoint
    And the user has marketplace suggestions disabled
    Then return 0 themes

Scenario: 3PD Extensibility
    Given there is a REST API endpoint
    And the response should be filterable to allow the theme collection to be customized by third parties.
    And the response should be filterable to allow the "Browse All" URL to be customized by third parties.
    
Scenario: Browse All URL
    Given there is a REST API endpoint
    And the "Browse All" URL should be provided in the _links response

```

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Make a request to GET `/wp-json/wc-admin/onboarding/themes/recommended`
3. It should return the following response.

```json
{
  "themes": [
    {
      "name": "Tsubaki",
      "price": "Free",
      "color_palettes": [],
      "slug": "tsubaki",
      "is_active": false,
      "thumbnail_url": "https://i0.wp.com/s2.wp.com/wp-content/themes/premium/tsubaki/screenshot.png",
      "link_url": "https://wordpress.com/theme/tsubaki/"
    },
    {
      "name": "Tazza",
      "price": "Free",
      "color_palettes": [],
      "slug": "tazza",
      "is_active": false,
      "thumbnail_url": "https://i0.wp.com/s2.wp.com/wp-content/themes/premium/tazza/screenshot.png",
      "link_url": "https://wordpress.com/theme/tazza/"
    },
    {
      "name": "Amulet",
      "price": "Free",
      "color_palettes": [],
      "slug": "amulet",
      "is_active": false,
      "thumbnail_url": "https://i0.wp.com/s2.wp.com/wp-content/themes/premium/amulet/screenshot.png",
      "link_url": "https://wordpress.com/theme/tsubaki/"
    },
    {
      "name": "Zaino",
      "price": "Free",
      "color_palettes": [],
      "slug": "zaino",
      "is_active": false,
      "thumbnail_url": "https://i0.wp.com/s2.wp.com/wp-content/themes/premium/zaino/screenshot.png",
      "link_url": "https://wordpress.com/theme/zaino/"
    }
  ],
  "_links": {
    "browse_all": {
      "href": "http://localhost:8885/mywordpress/wp-admin/themes.php"
    }
  }
}
```

4. Go to `WooCommerce > Settings > Advanced > WooCommerce.com` (/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com)
5. Uncheck `Display suggestions within WooCommerce` option
6. Repeat step 2
7.  It should return the following response.

```json
{
  "themes": [],
  "_links": {
    "browse_all": {
      "href": "http://localhost:8885/mywordpress/wp-admin/themes.php"
    }
  }
}
```

8. Add the following filter

```php
add_filter(
	'__experimental_woocommerce_rest_get_recommended_themes',
	function ( $result, $industry, $currency ) {
		$result['themes'] = array(
			'name'           => 'Tsubaki',
			'price'          => 'Free',
			'color_palettes' => array(),
			'slug'           => 'tsubaki',
			'is_active'      => false,
			'thumbnail_url'  => 'https://i0.wp.com/s2.wp.com/wp-content/themes/premium/tsubaki/screenshot.png',
			'link_url'       => 'https://wordpress.com/theme/tsubaki/',
		);
		return $result;
	},
	10,
	3,
);
```

7. Repeat step 2
8.  It should return the following response.

```json
{
  "themes": {
    "name": "Tsubaki",
    "price": "Free",
    "color_palettes": [],
    "slug": "tsubaki",
    "is_active": false,
    "thumbnail_url": "https://i0.wp.com/s2.wp.com/wp-content/themes/premium/tsubaki/screenshot.png",
    "link_url": "https://wordpress.com/theme/tsubaki/"
  },
  "_links": {
    "browse_all": {
      "href": "http://localhost:8885/mywordpress/wp-admin/themes.php"
    }
  }
}
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by psealock — COMMENTED** [review]
This is working well, however when I add the filter I'm not getting the theme to show up. Maybe something is off in my testing?

1. Fire up a JN site
2. Add this code to Hello Dolly (`plugins/hello.php`) and activate Hello Dolly
```
add_filter(
	'woocommerce_rest_get_recommended_themes',
	function ( $result, $industry, $currency ) {
		$result['themes'] = array(
			'name'           => 'Tsubaki',
			'price'          => 'Free',
			'color_palettes' => array(),
			'slug'           => 'tsubaki',
			'is_active'      => false,
			'thumbnail_url'  => 'https://i0.wp.com/s2.wp.com/wp-content/themes/premium/tsubaki/screenshot.png',
			'link_url'       => 'https://wordpress.com/theme/tsubaki/',
		);
		return $result;
	},
	10,
	3,
);
```
3. Refire the request in the console:
```js
await wp.apiFetch({
    path: '/wc-admin/onboarding/themes/recommended'
});
```
`themes` is an empty array.

File: plugins/woocommerce/src/Admin/API/OnboardingThemes.php [hunk 1]
```diff
@@ -184,6 +207,123 @@ public function activate_theme( $request ) {
...
+		 *
+		 * @param array $response The recommended themes response.
+		 * @param array $filtered_themes The filtered themes.
+		 * @param string $industry The industry to filter by (if provided).
+		 * @param string $currency The currency to convert prices to. (USD, AUD, CAD, EUR, GBP).
+		 *
+		 * @return array
+		 */
+		return apply_filters(
+			'woocommerce_rest_get_recommended_themes',
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Admin/API/OnboardingThemes.php]: Could we make this an experimental filter for now?

```suggestion
			'__experimental_woocommerce_rest_get_recommended_themes',
```

**Comment by adrianduffell** [comment]
> This is working well, however when I add the filter I'm not getting the theme to show up. Maybe something is off in my testing?

I think this might occur when Marketplace Suggestions is disabled. We could add the filter on the first `return` statement too.

**Comment by chihsuan** [comment]
> This is working well, however when I add the filter I'm not getting the theme to show up. Maybe something is off in my testing?

Thanks for the review! @psealock Yeah, it's because Marketplace Suggestions is off. 😅 

>  I think this might occur when Marketplace Suggestions is disabled. We could add the filter on the first return statement too.

Thanks @adrianduffell Added the filter in fba3b839181446f8e1799bd0d6f45c1a1fb4ba2a.

File: plugins/woocommerce/src/Admin/API/OnboardingThemes.php (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce/src/Admin/API/OnboardingThemes.php]: Sure! Updated in https://github.com/woocommerce/woocommerce/commit/fba3b839181446f8e1799bd0d6f45c1a1fb4ba2a.

**Review by adrianduffell — APPROVED** [review]
LGTM! The filter tested well now with marketplace suggestions both enabled and disabled.

I think we could merge it now to unblock further development on the intro screen 🚀

**Pull request merged**